File: JavaScript/2_Bucles/Ciclo_Ejercicio_9.js
/*
Una compañia de seguros tiene contratados a n vendedores. Cada uno hace tres ventas a la semana
Su politica de pagos es que cada vendedor recibe un sueldo base y el 10% de comision sobre sus ventas.
El gerente de la compañia quiere saber cuanto dinero obtendra a la semana cada vendedor por concepto
de comisiones por las tres ventas realizadas y cuanto tomando en cuenta su sueldo base y sus comisiones
*/

let numeroVendedores
//se define la variable que sera la condicion del bucle do while
let i = 1;
let ventas = 0;

let salario
let comisionSemana = 0;
let comisionAcumulada = 0;
let salir = true;

do{
    salario = parseFloat(prompt('Ingrese el salario base de los vendedores'));
    if(salario <= 0){
        alert('Error: El salario base debe ser mayor que cero.')
    }else{
        salir = false;
    }
}while(salir)

do{
    numeroVendedores = parseFloat(prompt('Ingrese el numero de vendedores: '));
    if(!Number.isInteger(numeroVendedores) || numeroVendedores < 1){
        alert('Error: Ingrese un numero valido de vendedores');
    }else{
        for(var j = 1; j <= numeroVendedores; j++){
            for(var k = 1; k <= 4; k++){
                ventas = parseFloat(prompt('Ingrese el valor de las tres ventas semana ' + k  + ' del vendedor ' + j))
            if(ventas < 0){
                alert('Error: Ingrese un valor valido de comision:');
            }else{
                comisionSemana = 0.1 * ventas;
                comisionAcumulada += comisionSemana;

                console.log('EL valor de la comision del vendedor ' + j + ' de la semana ' + k + ' es: ' + comisionSemana);
            }
            }
            salario = salario + comisionAcumulada;
            console.log('La comision del mes del vendedor'+ j +' es: ' + comisionAcumulada)
            console.log('EL valor del salario más comision del vendedor ' + j + ' de esta semana es: ' + salario);
            comisionAcumulada = 0;
            

        }
    }

    i++;
    console.log('Valor de variable bandera: ' + i);
}while(  i < numeroVendedores )


console.log('Numero Vendedores' + numeroVendedores);

File: Python/2_Bucles/Ciclo_Ejercicio_9.py
# Una compañia de seguros tiene contratados a n vendedores. Cada uno hace tres ventas a la semana
# Su politica de pagos es que cada vendedor recibe un sueldo base
#  y el 10% de comision sobre sus ventas.
# El gerente de la compañia quiere saber cuanto dinero obtendra a la semana cada vendedor
#  por concepto
# de comisiones por las tres ventas realizadas y cuanto tomando en cuenta su sueldo base
#  y sus comisiones

#Definicion de varaibles
numeroVendedores = 0
contadorSemanas = 0
ventaPorSemanaConComision = 0
entrar = True

#Ingresandoy validando el numero de vendedores.
while entrar:
    numeroVendedores = float(input('Ingrese el numero de vendedores: '))
    if numeroVendedores <= 0:
        print('Error: El numero de vendedores debe ser positivo:')
    else:
        sueldoBase = float(input('Ingrese el sueldo base: '))
        if sueldoBase < 0:
            print('Error: Valor invalido, ingrese un sueldo positivo')
        else:
            entrar = False

        
numeroVendedores = int(numeroVendedores)
print('')
#Calculando el sueldo:
for i  in range(0,numeroVendedores) : # itereaciones por vendedor
    print('Trabajador ',i +1)
    for j in range(1,5): #iteraciones por semana
        ventaPorSemana = float(input('Ingrese el valor de las 3 ventas semanales de la semana ' +  str(j) +' : '))
        if ventaPorSemana < 0:
            print('Error: Ingrese un valor de ventas mayor o igual a cero')
        else:
            ventaPorSemanaConComision += ventaPorSemana #Acumulando las ventas del mes   
            valorComisionSemanal = 0.1 * ventaPorSemana #calcula la comision en la semana
            print('La comision de la semana',j,'es:',valorComisionSemanal,'pesos')
            sueldoSemanal = sueldoBase / 4 +valorComisionSemanal
            print('El sueldo de la semana', j , 'es: ', sueldoSemanal, 'pesos')
        
    #Imprimiendo la camision y el sueldo del mes
    sueldoMensual = sueldoBase + ventaPorSemanaConComision
    print('Trabajador', i+1,':')
    print('Comision mensual:', ventaPorSemanaConComision, 'pesos' )
    print('Sueldo mensual del trabajor', j , ":",sueldoMensual )
    print('')
    


